scanpy/get.py

```diff
@@ -64,7 +74,22 @@ def rank_genes_groups_df(
         d = d[d["logfoldchanges"] < log2fc_max]
     if gene_symbols is not None:
         d = d.join(adata.var[gene_symbols], on="names")
-    return d
+
+    for pts, name in {'pts': 'pct_nz_group', 'pts_rest': 'pct_nz_reference'}.items():
+        if pts in adata.uns[key]:
+            pts_df = (
+                adata.uns[key][pts][group]
+                .reset_index()
+                .melt(id_vars='index', var_name='group', value_name=name)
+                .rename(columns={'index': 'names'})
+            )
```

Comment:
```suggestion
            pts_df = (
                adata.uns[key][pts][group]
                .rename_axis(index='names')
                .reset_index()
                .melt(id_vars='names', var_name='group', value_name=name)
            )
```


The `adata.var_names.name` won't always be index. It can be any string or `None`. 

For example:

```python
import scanpy as sc
pbmc = sc.datasets.pbmc3k_processed().raw.to_adata()
pbmc.var_names.name = "genes"
sc.tl.rank_genes_groups(pbmc, groupby="louvain", pts=True)

sc.get.rank_genes_groups_df(pbmc, group=None)
# KeyError: 'index'
```

<details>
<summary> Full traceback </summary>

```pytb
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/usr/local/lib/python3.8/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2894             try:
-> 2895                 return self._engine.get_loc(casted_key)
   2896             except KeyError as err:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

KeyError: 'index'

The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
<ipython-input-42-29eeffdb8753> in <module>
----> 1 sc.get.rank_genes_groups_df(pbmc, group=None)

~/github/scanpy/scanpy/get.py in rank_genes_groups_df(adata, group, key, pval_cutoff, log2fc_min, log2fc_max, gene_symbols)
     79         if pts in adata.uns[key]:
     80             pts_df = (
---> 81                 adata.uns[key][pts][group]
     82                 .reset_index()
     83                 .melt(id_vars='index', var_name='group', value_name=name)

/usr/local/lib/python3.8/site-packages/pandas/core/frame.py in melt(self, id_vars, value_vars, var_name, value_name, col_level, ignore_index)
   7164     ) -> "DataFrame":
   7165 
-> 7166         return melt(
   7167             self,
   7168             id_vars=id_vars,

/usr/local/lib/python3.8/site-packages/pandas/core/reshape/melt.py in melt(frame, id_vars, value_vars, var_name, value_name, col_level, ignore_index)
    120     mdata = {}
    121     for col in id_vars:
--> 122         id_data = frame.pop(col)
    123         if is_extension_array_dtype(id_data):
    124             id_data = cast("Series", concat([id_data] * K, ignore_index=True))

/usr/local/lib/python3.8/site-packages/pandas/core/frame.py in pop(self, item)
   4369         3  monkey        NaN
   4370         """
-> 4371         return super().pop(item=item)
   4372 
   4373     @doc(NDFrame.replace, **_shared_doc_kwargs)

/usr/local/lib/python3.8/site-packages/pandas/core/generic.py in pop(self, item)
    659 
    660     def pop(self, item: Label) -> Union["Series", Any]:
--> 661         result = self[item]
    662         del self[item]
    663         if self.ndim == 2:

/usr/local/lib/python3.8/site-packages/pandas/core/frame.py in __getitem__(self, key)
   2904             if self.columns.nlevels > 1:
   2905                 return self._getitem_multilevel(key)
-> 2906             indexer = self.columns.get_loc(key)
   2907             if is_integer(indexer):
   2908                 indexer = [indexer]

/usr/local/lib/python3.8/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2895                 return self._engine.get_loc(casted_key)
   2896             except KeyError as err:
-> 2897                 raise KeyError(key) from err
   2898 
   2899         if tolerance is not None:

KeyError: 'index'
```

</details>

It looks like if the index is not named, the column added is named `"index"`. However, if someone named one of their groups "index", the column would be named `"level_0"`. If there are columns named `"level_0"` and `"index"` pandas throws an error 😄. Either way, I think my suggestion should skip all this, unless someone named a group `"names"`.

This should probably get a test case.